RecoLocalCalo/HGCalRecProducers/plugins/HGCalUncalibRecHitWorkerWeights.cc

```diff
@@ -0,0 +1,60 @@
+#include "RecoLocalCalo/HGCalRecProducers/plugins/HGCalUncalibRecHitWorkerWeights.h"
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+HGCalUncalibRecHitWorkerWeights::HGCalUncalibRecHitWorkerWeights(const edm::ParameterSet&ps) :
+        HGCalUncalibRecHitWorkerBaseClass(ps)
+{
+}
+
+void
+HGCalUncalibRecHitWorkerWeights::set(const edm::EventSetup& es)
+{
+
+}
+
+
+bool
+HGCalUncalibRecHitWorkerWeights::run1( const edm::Event & evt,
+                const HGCEEDigiCollection::const_iterator & itdg,
+                HGCeeUncalibratedRecHitCollection & result )
+{
+        DetId detid(itdg->id());
+
+        if (detid.subdetId()==HGCEE) {
```

Comment:
this paranoid check may be not needed eventually.
The function signature already suggests that the input is HGCEE.